src/main/kotlin/net/alexwells/kottle/FMLKotlinModContainer.kt

```diff
@@ -30,102 +29,89 @@ import java.util.function.Supplier
+            }
+        } catch (e: Throwable) {
+            logger.error(LOADING, "Failed to load class $className", e)
+            throw ModLoadingException(modInfo, ModLoadingStage.CONSTRUCT, "fml.modloading.failedtoloadmodclass", e)
+        }
+        // Then we check whether it's a kotlin object and return it, or if not we create a new instance of kotlin class.
         try {
-            logger.debug(LOADING, "Loading mod instance {} of type {}", getModId(), modClass.name)
-            // Now we can load the class, so that static {} block gets called
-            Class.forName(className) // todo: use modClassLoader? IMPORTANT
-            // Then we check whether it's a kotlin object and return it, or if not we create a new instance of kotlin class.
+            logger.debug(LOADING, "Loading mod instance ${getModId()} of type $className")
             this.mod = modClass.kotlin.objectInstance ?: modClass.getConstructor().newInstance()
-            logger.debug(LOADING, "Loaded mod instance {} of type {}", getModId(), modClass.name)
+            logger.debug(LOADING, "Loaded mod instance ${getModId()} of type $className")
         } catch (e: Throwable) {
             if (e is IllegalStateException) {
-                logger.error(
-                        LOADING,
-                        "This seems like a compatibility error between Kottle and the mod. " +
-                                "Please make sure identical versions of shadowed libraries are used in the mod and Kottle of this version."
+                logger.error(LOADING, "This seems like a compatibility error between Kottle and the mod. " +
+                        "Please make sure identical versions of shadowed libraries are used in the mod and Kottle of this version."
                 )
             }
 
-            logger.error(LOADING, "Failed to create mod instance. ModID: {}, class {}", getModId(), modClass.name, e)
+            logger.error(LOADING, "Failed to create mod instance. ModID: ${getModId()}, class $className", e)
 
             throw ModLoadingException(modInfo, event.fromStage(), "fml.modloading.failedtoloadmod", e, modClass)
         }
 
-        logger.debug(LOADING, "Injecting Automatic event subscribers for {}", getModId())
-        KotlinAutomaticEventSubscriber.inject(this, this.scanResults, this.modClass.classLoader)
-        logger.debug(LOADING, "Completed Automatic event subscribers for {}", getModId())
+        logger.debug(LOADING, "Injecting Automatic event subscribers for ${getModId()}")
+        KotlinAutomaticEventSubscriber.inject(this, scanResults, modClass.classLoader)
+        logger.debug(LOADING, "Completed Automatic event subscribers for ${getModId()}")
     }
 
-    private fun dummy(): Consumer<LifecycleEventProvider.LifecycleEvent> = Consumer {}
-
-    private fun beforeEvent(lifecycleEvent: LifecycleEventProvider.LifecycleEvent) {}
+    private val dummy get() = Consumer<LifecycleEventProvider.LifecycleEvent> {}
```

Comment:
Not sure about this. This is a factory, while I would expect a getter to always return the same reference.

Reply:
Nope, that's a variable with a custom getter. There isn't an underlying field there. When compiled there's no actual variable there, just a `getDummy()` function which looks identical to the `dummy()` function.